torch/csrc/jit/python/python_tracer.cpp

```diff
@@ -21,20 +21,39 @@ namespace torch {
 namespace jit {
 namespace tracer {
 
+namespace {
+const std::string kPySelf = "self";
+}
+
 // Python interpreter retrieval routine adapted from
 // https://stackoverflow.com/a/8706144
-std::vector<StackEntry> _pythonCallstack() {
+std::vector<FileLineFunc> _pythonCallstack() {
   pybind11::gil_scoped_acquire gil;
   PyFrameObject* frame = PyEval_GetFrame();
-  std::vector<StackEntry> entries;
+  std::vector<FileLineFunc> entries;
 
-  while (nullptr != frame) {
+  while (frame != nullptr) {
     size_t line = PyCode_Addr2Line(frame->f_code, frame->f_lasti);
     std::string filename = THPUtils_unpackString(frame->f_code->co_filename);
     std::string funcname = THPUtils_unpackString(frame->f_code->co_name);
-    auto source = std::make_shared<Source>(funcname, filename, line);
+    std::string classname = "";
+
+    if (PyFrame_FastToLocalsWithError(frame) >= 0) {
```

Comment:
you probably want to trigger it only when you encounter the frame that belongs to torch.nn.Module._call_impl. You can do it by looking up the code object for that method once and then just compare it with `frame->f_code`. That's how https://colab.research.google.com/drive/1NGRzvgzCZR6EEM3RVPvPAuMXfHsyC6eN?usp=sharing worked

Reply:
yeah, I went with all class names, not only Module specific; we could limit this to Module-inherited ones